node/src/checkpointer.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface Checkpointer {
+    checkpoint(blockNumber: bigint, transactionId?: string): Promise<void>;
+    getBlockNumber(): bigint | undefined;
+    getTransactionIds(): string[];
```

Comment:
Just a reminder, large numbers of transactions can be much more efficiently located with a Set rather than linear array search:
```suggestion
    getTransactionIds(): Set<string>;
```